jbpm-test-coverage/src/test/java/org/jbpm/test/functional/event/BoundaryEventWithOutputSetPersistenceTest.java

```diff
@@ -0,0 +1,415 @@
+        int processState = process.getState();
+        assertEquals("Expected process state to be " + processStateName[ProcessInstance.STATE_ACTIVE] + " not "
+                + processStateName[processState], ProcessInstance.STATE_ACTIVE, processState);
+
+        completeWork(ksession, humanTaskMockHandler);
+        ksession.signalEvent("Compensation", "_2");
+        
+        
+        completeWork(ksession, humanTaskMockHandler);
+        // The process completes
+        process = ksession.getProcessInstance(process.getId());
+        assertNull("Expected process to have been completed and removed", process);
+        assertNull(dataTaskMockHandler.getInputParameter("nodeInstance"));
+        assertEquals(dataTaskMockHandler.getInputParameter("event"), "_2");
+        assertEquals(dataTaskMockHandler.getInputParameter("signal"), "Compensation");
+    }
+    
+    @Test
+    public void testBoundaryConditionalSignalAndCompleteHumanTask() throws InterruptedException {
+        
+        createRuntimeManager(PROCESS_CONDITIONAL_FILE_NAME);
+
+        RuntimeEngine runtimeEngine = getRuntimeEngine();
+        KieSession ksession = runtimeEngine.getKieSession();
+        long ksessionId = ksession.getIdentifier();
+        assertTrue("session id not saved.", ksessionId > 0);
+        
+        HumanTaskMockHandler humanTaskMockHandler = new HumanTaskMockHandler();
+        DataTaskMockHandler dataTaskMockHandler = new DataTaskMockHandler();
+        // Register Human Task Handler
+        ksession.getWorkItemManager().registerWorkItemHandler(WORK_ITEM_HANDLER_TASK, humanTaskMockHandler);
+        ksession.getWorkItemManager().registerWorkItemHandler(WORK_ITEM_HANDLER_DATA, dataTaskMockHandler);
+        // Start the process 
+        ProcessInstance process = ksession.startProcess(PROCESS_CONDITIONAL_NAME);
+        long processId = process.getId();
+        assertTrue("process id not saved", processId > 0);
+        
+        // The process is in the Human Task waiting for its completion
+        int processState = process.getState();
+        assertEquals("Expected process state to be " + processStateName[ProcessInstance.STATE_ACTIVE] + " not "
+                + processStateName[processState], ProcessInstance.STATE_ACTIVE, processState);
+
+        ksession.signalEvent("RuleFlowStateEvent-" + PROCESS_CONDITIONAL_NAME + "-4-_2" , new org.jbpm.test.entity.Person(1L, "john"));
+       
+        completeWork(ksession, humanTaskMockHandler);
+
+        // The process completes
+        process = ksession.getProcessInstance(process.getId());
+        assertNull("Expected process to have been completed and removed", process);
+        assertTrue(dataTaskMockHandler.getInputParameter("nodeInstance") instanceof HumanTaskNodeInstance);
+        assertTrue(dataTaskMockHandler.getInputParameter("event") instanceof Person);
+        assertEquals(dataTaskMockHandler.getInputParameter("signal"), "RuleFlowStateEvent-BoundaryConditionalEventProcess-4-_2");
```

Comment:
Shouldn't we check `workItem` here as well?

Reply:
done